src/js/main.js

```diff
@@ -65,27 +65,28 @@
         var attribute = mp.prefs.get('attribute');
 
         var url = new URL('v1/contextEntities/type/' + entity_type + '/id/' + entity + '/attributes/' + attribute, server);
+        
         mp.http.makeRequest(url, {
             method: "GET",
             requestHeaders: request_headers,
             parameters: {
                 lastN: HLIMIT,
                 dateFrom: from.toISOString()/* ,
                 dateTo: to.toISOString()*/
-            }
-        }).then(function (response) {
-            if (response.status !== 200) {
-                throw new Error('Unexpected response from STH');
-            }
-
-            // Current version of STH only provides a contextResponse and only a attribute entry
-            var data = JSON.parse(response.responseText).contextResponses[0].contextElement.attributes[0].values;
-            mp.wiring.pushEvent("values", data.map(function (entry) {return Number(entry.attrValue);}));
-            mp.wiring.pushEvent("timestamps", data.map(function (entry) {return (new Date(entry.recvTime)).getTime();}));
-        }).catch(function (reason) {
-            mp.operator.log(reason);
+            },
+                onSuccess: function (response) {
+                        var data = JSON.parse(response.responseText).contextResponses[0].contextElement.attributes[0].values;
```

Comment:
`onSuccess` only checks if the status code returned by the server is in the 2xx range. Probably this method can be improved by checking if the response uses the expected status code:

```javascript
            if (response.status !== 200) {
                throw new Error('Unexpected response from STH');
            }
```